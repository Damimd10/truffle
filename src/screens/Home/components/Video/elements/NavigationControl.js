import styled from 'styled-components';

const NavigationControl = styled.section`
  display: flex;
  flex-wrap: wrap;
  margin: 0.5rem;
  justify-content: flex-end;
  align-items: center;
  flex-basis: 500px;
`;

NavigationControl.displayName = 'NavigationControlContainer';

export default NavigationControl;
