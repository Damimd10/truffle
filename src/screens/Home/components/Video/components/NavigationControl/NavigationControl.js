import React from 'react';
import { string } from 'prop-types';
import { Link } from 'react-router-dom';
import { isEmpty } from 'ramda';

import { Button, NavigationControlContainer } from '../../elements';

const NavigationControl = ({ prevUrl, nextUrl }) => (
  <NavigationControlContainer>
    <Link to={prevUrl}>
      <Button type="button" disabled={isEmpty(prevUrl)}>
        Prev Video
      </Button>
    </Link>
    <Link to={nextUrl}>
      <Button type="button" disabled={isEmpty(nextUrl)}>
        Next Video
      </Button>
    </Link>
  </NavigationControlContainer>
);

NavigationControl.propTypes = {
  prevUrl: string.isRequired,
  nextUrl: string.isRequired,
};

export default NavigationControl;
