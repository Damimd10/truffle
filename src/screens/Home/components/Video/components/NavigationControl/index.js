import NavigationControl from './NavigationControl';

export default NavigationControl;
