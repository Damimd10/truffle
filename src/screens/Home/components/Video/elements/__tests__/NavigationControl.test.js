import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import NavigationControl from '../NavigationControl';

describe('Button Style Element', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NavigationControl />);
  });

  it('should create a style structure for the component', () =>
    expect(toJson(wrapper)).toMatchSnapshot());
});
