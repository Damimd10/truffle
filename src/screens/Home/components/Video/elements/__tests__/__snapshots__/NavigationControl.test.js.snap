// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Style Element should create a style structure for the component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-wrap: wrap;
  margin: 0.5rem;
  justify-content: flex-end;
  align-items: center;
  flex-basis: 500px;
",
        ],
      },
      "displayName": "NavigationControlContainer",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "section",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
